packages/@aws-cdk/aws-appmesh/lib/tls-certificate.ts

```diff
@@ -0,0 +1,171 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualNode } from './appmesh.generated';
+
+/**
+ * Enum of supported TLS modes
+ */
+export enum TlsMode {
+  /**
+   * Only accept encrypted traffic
+   */
+  STRICT = 'STRICT',
+
+  /**
+   * Accept encrypted and plaintext traffic.
+   */
+  PERMISSIVE = 'PERMISSIVE',
+
+  /**
+   * TLS is disabled, only accept plaintext traffic.
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * A wrapper for the tls config returned by {@link TlsCertificate.bind}
+ */
+export interface TlsCertificateConfig {
+  /**
+   * The CFN shape for a listener TLS certificate
+   */
+  readonly tlsCertificate: CfnVirtualNode.ListenerTlsCertificateProperty,
+}
+
+/**
+ * ACM Certificate Properties
+ */
+export interface AcmCertificateOptions {
+  /**
+   * The TLS mode.
+   */
+  readonly tlsMode: TlsMode;
+
+  /**
+   * The ACM certificate
+   */
+  readonly acmCertificate: acm.ICertificate;
+}
+
+/**
+ * File Certificate Properties
+ */
+export interface FileCertificateOptions {
+  /**
+   * The TLS mode.
+   */
+  readonly tlsMode: TlsMode;
+
+  /**
+   * The file path of the certificate chain file.
+   */
+  readonly certificateChain: string;
+
+  /**
+   * The file path of the private key file.
+   */
+  readonly privateKey: string;
+}
```

Comment:
I'm wondering whether `certificateChainPath` and `privateKeyPath` are better names here (to make it absolutely clear these are file paths).

Reply:
I named these to have parity with the FE API, but I agree it disambiguates the variable name.